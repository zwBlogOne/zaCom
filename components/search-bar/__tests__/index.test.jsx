import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBar from '../index';

describe('SearchBar', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <SearchBar
        shape="round"
        cancelText="取消"
        placeholder="搜索"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders defaultValue correctly', () => {
    const wrapper = mount(
      <SearchBar
        shape="round"
        cancelText="取消取消"
        placeholder="搜索"
      />
    );
    wrapper.setProps({ defaultValue: '搜索关键字' });
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders onFocus called correctly', () => {
    const onFocus = jest.fn();
    const wrapper = mount(
      <SearchBar
        shape="round"
        placeholder="搜索"
        onFocus={onFocus}
      />
    );
    wrapper.find('input[type="search"]').simulate('focus');
    expect(onFocus).toBeCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders onChange called correctly', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <SearchBar
        shape="round"
        placeholder="搜索"
        onChange={onChange}
      />
    );

    const input = wrapper.find('input[type="search"]');
    input.simulate('change', { target: { value: '测试值' } });
    expect(input.instance().value).toEqual('测试值');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('renders onBlur called correctly', () => {
  //   const onBlur = jest.fn();
  //   const wrapper = render(
  //     <SearchBar
  //       shape="round"
  //       placeholder="搜索"
  //       onBlur={onBlur}
  //     />
  //   );

  //   const input = wrapper.find('input[type="search"]');
  //   // const spyOnBlur = jest.spyOn(wrapper.instance(), 'onBlur');
  //   input.props('onBlur')();
  //   expect(onBlur).toHaveBeenCalled();
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });

  it('renders onCancel called correctly', () => {
    const onCancel = jest.fn();
    const wrapper = mount(
      <SearchBar
        shape="round"
        placeholder="搜索"
        onCancel={onCancel}
      />
    );
    const input = wrapper.find('input[type="search"]');
    input.simulate('focus');
    wrapper.find('.za-searchbar-cancel').simulate('click');
    expect(onCancel).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders onSubmit called correctly', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(
      <SearchBar
        shape="round"
        placeholder="搜索"
        onSubmit={onSubmit}
      />
    );

    const input = wrapper.find('input[type="search"]');
    input.simulate('change', { target: { value: 'My new value' } });
    wrapper.find('.za-searchbar-form').simulate('submit');
    expect(onSubmit).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
