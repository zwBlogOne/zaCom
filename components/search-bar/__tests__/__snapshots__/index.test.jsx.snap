// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  class="za-searchbar"
>
  <form
    action="#"
    class="za-searchbar-form"
  >
    <div
      class="za-searchbar-content shape-round"
    >
      <div
        class="za-searchbar-mock"
      >
        <div
          class="za-searchbar-mock-container"
        >
          <i
            class="za-icon za-icon-search"
          />
          <span
            class="za-searchbar-mock-placeholder"
            style="visibility:visible"
          >
            搜索
          </span>
        </div>
      </div>
      <div
        class="za-input za-input-search clearable"
      >
        <input
          autocomplete="off"
          placeholder="搜索"
          type="search"
          value=""
        />
        <i
          class="za-icon za-input-clear za-icon-wrong-round-fill"
        />
      </div>
    </div>
    <div
      class="za-searchbar-cancel"
    >
      取消
    </div>
  </form>
</div>
`;

exports[`SearchBar renders defaultValue correctly 1`] = `
<SearchBar
  cancelText="取消取消"
  clearable={true}
  defaultValue="搜索关键字"
  disabled={false}
  placeholder="搜索"
  prefixCls="za-searchbar"
  shape="round"
  showCancel={false}
>
  <div
    className="za-searchbar"
  >
    <form
      action="#"
      className="za-searchbar-form"
      onSubmit={[Function]}
    >
      <div
        className="za-searchbar-content shape-round"
      >
        <div
          className="za-searchbar-mock"
        >
          <div
            className="za-searchbar-mock-container"
          >
            <Icon
              prefixCls="za-icon"
              type="search"
            >
              <i
                className="za-icon za-icon-search"
              />
            </Icon>
            <span
              className="za-searchbar-mock-placeholder"
              style={
                Object {
                  "visibility": "visible",
                }
              }
            >
              搜索
            </span>
          </div>
        </div>
        <InputBase
          clearable={true}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClear={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onCompositionUpdate={[Function]}
          onFocus={[Function]}
          placeholder="搜索"
          prefixCls="za-input"
          type="search"
          value=""
        >
          <div
            className="za-input za-input-search clearable"
          >
            <input
              autoComplete="off"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onCompositionUpdate={[Function]}
              onFocus={[Function]}
              placeholder="搜索"
              type="search"
              value=""
            />
            <Icon
              className="za-input-clear"
              onClick={[Function]}
              prefixCls="za-icon"
              type="wrong-round-fill"
            >
              <i
                className="za-icon za-input-clear za-icon-wrong-round-fill"
                onClick={[Function]}
              />
            </Icon>
          </div>
        </InputBase>
      </div>
      <div
        className="za-searchbar-cancel"
        onClick={[Function]}
      >
        取消取消
      </div>
    </form>
  </div>
</SearchBar>
`;

exports[`SearchBar renders onCancel called correctly 1`] = `
<SearchBar
  cancelText="取消"
  clearable={true}
  disabled={false}
  onCancel={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  placeholder="搜索"
  prefixCls="za-searchbar"
  shape="round"
  showCancel={false}
>
  <div
    className="za-searchbar"
  >
    <form
      action="#"
      className="za-searchbar-form"
      onSubmit={[Function]}
    >
      <div
        className="za-searchbar-content shape-round"
      >
        <div
          className="za-searchbar-mock"
        >
          <div
            className="za-searchbar-mock-container"
          >
            <Icon
              prefixCls="za-icon"
              type="search"
            >
              <i
                className="za-icon za-icon-search"
              />
            </Icon>
            <span
              className="za-searchbar-mock-placeholder"
              style={
                Object {
                  "visibility": "visible",
                }
              }
            >
              搜索
            </span>
          </div>
        </div>
        <InputBase
          clearable={true}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClear={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onCompositionUpdate={[Function]}
          onFocus={[Function]}
          placeholder="搜索"
          prefixCls="za-input"
          type="search"
          value=""
        >
          <div
            className="za-input za-input-search clearable"
          >
            <input
              autoComplete="off"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onCompositionUpdate={[Function]}
              onFocus={[Function]}
              placeholder="搜索"
              type="search"
              value=""
            />
            <Icon
              className="za-input-clear"
              onClick={[Function]}
              prefixCls="za-icon"
              type="wrong-round-fill"
            >
              <i
                className="za-icon za-input-clear za-icon-wrong-round-fill"
                onClick={[Function]}
              />
            </Icon>
          </div>
        </InputBase>
      </div>
      <div
        className="za-searchbar-cancel"
        onClick={[Function]}
      >
        取消
      </div>
    </form>
  </div>
</SearchBar>
`;

exports[`SearchBar renders onChange called correctly 1`] = `
<SearchBar
  cancelText="取消"
  clearable={true}
  disabled={false}
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "测试值",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  placeholder="搜索"
  prefixCls="za-searchbar"
  shape="round"
  showCancel={false}
>
  <div
    className="za-searchbar"
  >
    <form
      action="#"
      className="za-searchbar-form za-searchbar-form-focus"
      onSubmit={[Function]}
    >
      <div
        className="za-searchbar-content shape-round"
      >
        <div
          className="za-searchbar-mock"
        >
          <div
            className="za-searchbar-mock-container"
          >
            <Icon
              prefixCls="za-icon"
              type="search"
            >
              <i
                className="za-icon za-icon-search"
              />
            </Icon>
            <span
              className="za-searchbar-mock-placeholder"
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              搜索
            </span>
          </div>
        </div>
        <InputBase
          clearable={true}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClear={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onCompositionUpdate={[Function]}
          onFocus={[Function]}
          placeholder="搜索"
          prefixCls="za-input"
          type="search"
          value="测试值"
        >
          <div
            className="za-input za-input-search clearable"
          >
            <input
              autoComplete="off"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onCompositionUpdate={[Function]}
              onFocus={[Function]}
              placeholder="搜索"
              type="search"
              value="测试值"
            />
            <Icon
              className="za-input-clear za-input-clear-show"
              onClick={[Function]}
              prefixCls="za-icon"
              type="wrong-round-fill"
            >
              <i
                className="za-icon za-input-clear za-input-clear-show za-icon-wrong-round-fill"
                onClick={[Function]}
              />
            </Icon>
          </div>
        </InputBase>
      </div>
      <div
        className="za-searchbar-cancel za-searchbar-cancel-show"
        onClick={[Function]}
      >
        取消
      </div>
    </form>
  </div>
</SearchBar>
`;

exports[`SearchBar renders onFocus called correctly 1`] = `
<SearchBar
  cancelText="取消"
  clearable={true}
  disabled={false}
  onFocus={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  placeholder="搜索"
  prefixCls="za-searchbar"
  shape="round"
  showCancel={false}
>
  <div
    className="za-searchbar"
  >
    <form
      action="#"
      className="za-searchbar-form za-searchbar-form-focus"
      onSubmit={[Function]}
    >
      <div
        className="za-searchbar-content shape-round"
      >
        <div
          className="za-searchbar-mock"
        >
          <div
            className="za-searchbar-mock-container"
          >
            <Icon
              prefixCls="za-icon"
              type="search"
            >
              <i
                className="za-icon za-icon-search"
              />
            </Icon>
            <span
              className="za-searchbar-mock-placeholder"
              style={
                Object {
                  "visibility": "visible",
                }
              }
            >
              搜索
            </span>
          </div>
        </div>
        <InputBase
          clearable={true}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClear={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onCompositionUpdate={[Function]}
          onFocus={[Function]}
          placeholder="搜索"
          prefixCls="za-input"
          type="search"
          value=""
        >
          <div
            className="za-input za-input-search clearable"
          >
            <input
              autoComplete="off"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onCompositionUpdate={[Function]}
              onFocus={[Function]}
              placeholder="搜索"
              type="search"
              value=""
            />
            <Icon
              className="za-input-clear"
              onClick={[Function]}
              prefixCls="za-icon"
              type="wrong-round-fill"
            >
              <i
                className="za-icon za-input-clear za-icon-wrong-round-fill"
                onClick={[Function]}
              />
            </Icon>
          </div>
        </InputBase>
      </div>
      <div
        className="za-searchbar-cancel za-searchbar-cancel-show"
        onClick={[Function]}
      >
        取消
      </div>
    </form>
  </div>
</SearchBar>
`;

exports[`SearchBar renders onSubmit called correctly 1`] = `
<SearchBar
  cancelText="取消"
  clearable={true}
  disabled={false}
  onSubmit={
    [MockFunction] {
      "calls": Array [
        Array [
          "My new value",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  placeholder="搜索"
  prefixCls="za-searchbar"
  shape="round"
  showCancel={false}
>
  <div
    className="za-searchbar"
  >
    <form
      action="#"
      className="za-searchbar-form za-searchbar-form-focus"
      onSubmit={[Function]}
    >
      <div
        className="za-searchbar-content shape-round"
      >
        <div
          className="za-searchbar-mock"
        >
          <div
            className="za-searchbar-mock-container"
          >
            <Icon
              prefixCls="za-icon"
              type="search"
            >
              <i
                className="za-icon za-icon-search"
              />
            </Icon>
            <span
              className="za-searchbar-mock-placeholder"
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              搜索
            </span>
          </div>
        </div>
        <InputBase
          clearable={true}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClear={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onCompositionUpdate={[Function]}
          onFocus={[Function]}
          placeholder="搜索"
          prefixCls="za-input"
          type="search"
          value="My new value"
        >
          <div
            className="za-input za-input-search clearable"
          >
            <input
              autoComplete="off"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onCompositionUpdate={[Function]}
              onFocus={[Function]}
              placeholder="搜索"
              type="search"
              value="My new value"
            />
            <Icon
              className="za-input-clear za-input-clear-show"
              onClick={[Function]}
              prefixCls="za-icon"
              type="wrong-round-fill"
            >
              <i
                className="za-icon za-input-clear za-input-clear-show za-icon-wrong-round-fill"
                onClick={[Function]}
              />
            </Icon>
          </div>
        </InputBase>
      </div>
      <div
        className="za-searchbar-cancel za-searchbar-cancel-show"
        onClick={[Function]}
      >
        取消
      </div>
    </form>
  </div>
</SearchBar>
`;
